src/funcs.rs

```diff
@@ -465,7 +465,7 @@ pub fn clean_msg(
             .unwrap();
         }
 
-        let said_name = nickname.unwrap_or_else(|| member_nick.unwrap_or(&user.name));
+        let said_name = nickname.unwrap_or_else(|| member_nick.unwrap_or_else(|| &user.global_name.unwrap_or(&user.name)));
```

Comment:
```suggestion
        let said_name = nickname.unwrap_or_else(|| member_nick.unwrap_or_else(|| &user.global_name.or(&user.name)));
```

you mean like that? sorry i'm not familiar enough

Reply:
Yeah, but for the other two unwrap_or_else calls aswell